integration-tests/manifests/annotations/validate_annotations.go

```diff
@@ -0,0 +1,752 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent-operator/pkg/instrumentation/auto"
+	appsV1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func main() {
+
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		fmt.Printf("error getting user home dir: %v\n\n", err)
+	}
+	kubeConfigPath := filepath.Join(userHomeDir, ".kube", "config")
+	fmt.Printf("Using kubeconfig: %s\n\n", kubeConfigPath)
+
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		fmt.Printf("Error getting kubernetes config: %v\n\n", err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(kubeConfig)
+
+	if err != nil {
+		fmt.Printf("error getting kubernetes config: %v\n\n", err)
+	}
+	success := verifyAutoAnnotation(clientSet)
+	if !success {
+		fmt.Println("Instrumentation Annotation Injection Test: FAIL")
+		os.Exit(1)
+	} else {
+		fmt.Println("Instrumentation Annotation Injection Test: PASS")
+	}
+}
+
+const injectJavaAnnotation = "instrumentation.opentelemetry.io/inject-java"
+const autoAnnotateJavaAnnotation = "cloudwatch.aws.amazon.com/auto-annotate-java"
+const injectPythonAnnotation = "instrumentation.opentelemetry.io/inject-python"
+const autoAnnotatePythonAnnotation = "cloudwatch.aws.amazon.com/auto-annotate-python"
+const defaultNamespace = "default"
+const deploymentName = "nginx"
+const amazonCloudwatchNamespace = "amazon-cloudwatch"
+
+const amazonControllerManager = "cloudwatch-controller-manager"
+
+func verifyAutoAnnotation(clientSet *kubernetes.Clientset) bool {
+
+	//updating operator deployment
+	deployment, err := clientSet.AppsV1().Deployments(amazonCloudwatchNamespace).Get(context.TODO(), amazonControllerManager, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Error getting deployment: %v\n\n", err)
+		os.Exit(1)
+	}
+
+	//---------------------------USE CASE 1 (Java and Python on Deployment) ----------------------------------------------
```

Comment:
We should avoid the behavior of one test bleeding into another.
We can make these all separate test cases that can then be run in parallel - just make sure each test case uses a dynamic namespace spun up for that particular test case so there is very little overlap.
Spinning up a new cluster per test case might be an overkill.

Reply:
Question: How do run the tests in parallel? Are we changing all the test to run concurrently through threads and locks?

Reply:
The reason why I did the bleed in behavior was mostly to test if there is a language already annotated but the user removes that language's annotation, than we should verify it removal.

Reply:
>The reason why I did the bleed in behavior was mostly to test if there is a language already annotated but the user removes that language's annotation, than we should verify it removal.

Yes that should be the behavior for one of the tests. But once that flow of events is done, it shouldnt impact any of the other tests.

Reply:
> How do run the tests in parallel?

You can look at `t.Parallel()`.